src/main/java/com/cleo/rest/services/impl/CarsWebService.java

```diff
@@ -1,30 +1,333 @@
+   * Adds a new car to the list of cars.
+   *
+   * @param car
+   * @return -> Response verifying that a car has been added to list of cars.
+   * @throws Exception
+   */
+  @Override
+  public Response addCar(Car car) throws Exception{
+    car.setId(UUID.randomUUID());
+    cars.add(car);
+    return Response.created(new URI("http://localhost:8080/api/cars/" + car.getId())).build();
+  }
+
+
+  /***
+   * Deletes car from list of cars as specified by id.
+   *
+   * @param id
+   * @return -> Response verifying that the car has been deleted from list of cars.
+   */
+  @Override
+  public Response deleteCarById(String id) {
+
+    if(Strings.isNullOrEmpty(id)) {
+      throw new InvalidIdException("ID cannot be null or empty", Response.Status.BAD_REQUEST);
+    }
+    try{
+      UUID.fromString(id);
+    }catch(Exception ex){
+      throw new InvalidIdException("ID is not a valid UUID", Response.Status.BAD_REQUEST);
+    }
+
+    Car myCar = findCar(id);
+    if(myCar == null) {
+      throw new InvalidIdException("No car with that ID", Response.Status.NOT_FOUND);
+    }
+
+    return Response.ok(
+        cars.remove(myCar) + ", " + myCar.getYear() + " " + myCar.getColor() + " " + myCar.getMake() + " "
+            + myCar.getModel() + " has been deleted!"
+    ).build();
+  }
+
+
+  /***
+   * Purchases car as specified from id, removing the car from the list of cars.
+   *
+   * @param id
+   * @return -> Response that car has been purchased and removed from list of cars.
+   */
+  @Override
+  public Response purchaseCarById(String id) {
+
+    if(Strings.isNullOrEmpty(id)) {
+      throw new InvalidIdException("ID cannot be null or empty", Response.Status.BAD_REQUEST);
+    }
+    try{
+      UUID.fromString(id);
+    }catch(Exception ex){
+      throw new InvalidIdException("ID is not a valid UUID", Response.Status.BAD_REQUEST);
+    }
+
+    Car myCar = findCar(id);
+    if(myCar == null) {
+      throw new InvalidIdException("No car with that ID", Response.Status.NOT_FOUND);
+    }
+
+    return Response.ok(
+        cars.remove(myCar) + ", " + myCar.getYear() + " " + myCar.getColor() + " " + myCar.getMake() + " "
+            + myCar.getModel() + " has been purchased!"
+    ).build();
+  }
+
+
+  /***
+   * Updates car by id, taking in a car with desired attributes to be updated.
+   *
+   * @param id
+   * @param car
+   * @return -> Returns the updated car as specified by id.
+   */
   @Override
-  public Response getAll() {
+  public Response updateCarById(String id, Car car) {
+
+    if(Strings.isNullOrEmpty(id)) {
+      throw new InvalidIdException("ID cannot be null or empty", Response.Status.BAD_REQUEST);
+    }
+    try{
+      UUID.fromString(id);
+    }catch(Exception ex){
+      throw new InvalidIdException("ID is not a valid UUID", Response.Status.BAD_REQUEST);
+    }
+
+    Car myCar = findCar(id);
+    if(myCar == null) {
+      throw new InvalidIdException("No car with that ID", Response.Status.NOT_FOUND);
+    }
```

Comment:
We have this same validation logic in all of the methods that require an id. Can you filter these checks out to some kind of helper method?

I think that will make a good candidate for our unit test once we get there.